### PR TITLE
ci:  Added step to check if @tag.All is given as expected.

### PR DESCRIPTION
**Pull request by sharanya-appsmith**
#### Type of change
- Chore (housekeeping or task changes that don't impact user perception)

#### How Has This Been Tested?
- [x] Manual
- [x] TBP
>

#### Dev activity
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
- [ ] PR is being merged under a feature flag

#### QA activity:
- [ ] [Speedbreak features](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#speedbreakers-) have been covered
- [ ] Test plan covers all impacted features and [areas of interest](https://github.com/appsmithorg/TestSmith/wiki/Guidelines-for-test-plans#areas-of-interest-)
- [ ] Test plan has been peer reviewed by project stakeholders and other QA members
- [ ] Manually tested functionality on DP
- [ ] We had an implementation alignment call with stakeholders post QA Round 2
- [ ] Cypress test cases have been added and approved by SDET/manual QA
- [ ] Added `Test Plan Approved` label after Cypress tests were reviewed
- [ ] Added `Test Plan Approved` label after JUnit tests were reviewed

## Summary by CodeRabbit

The existing bullet-point list is still valid and does not need to be updated.

### CODE REVIEW

File: .github/workflows/test-build-docker-image.yml [hunk 1]
```diff
@@ -33,9 +33,14 @@ jobs:
...
-              echo "tags=${{inputs.tags}}" >> $GITHUB_OUTPUT
-              echo "matrix=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]" >> $GITHUB_OUTPUT
+            if [[ "${{inputs.tags}}" != "" ]]; then
+              if [[ $tags == *"ALL"* || $tags == *"All"* && $tags != "@tag.All" ]]; then
+                echo "Please use `/ok-to-test tags="@tag.All"`" >> $GITHUB_OUTPUT
+                exit 1  
+              else
+                echo "tags=${{inputs.tags}}" >> $GITHUB_OUTPUT
+                echo "matrix=[0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10, 11, 12, 13, 14, 15, 16, 17, 18, 19]" >> $GITHUB_OUTPUT
+              fi
```
[the remark this answers is not in the document]
Comment (sharanya-appsmith) [reply .github/workflows/test-build-docker-image.yml]: Done

File: .github/workflows/integration-tests-command.yml [hunk 2]
```diff
@@ -36,7 +36,10 @@ jobs:
         id: checkAll
         run: |
           tags="${{ github.event.client_payload.slash_command.args.named.tags }}"
-          if [[ $tags == *"@tag.All"* ]]; then
+          if [[ $tags == *"ALL"* || $tags == *"All"* && $tags != "@tag.All" ]]; then
```
Comment (mohanarpit) [inline .github/workflows/integration-tests-command.yml:39]: This `if` condition seems to have some redundancy. The clause `$tags == *"ALL"*` is repeated in an OR condition.

**Comment by sarojsarab** [comment]
Add all the links to the runs where you tested this.

**Comment by sarojsarab** [comment]
Also don't see any comment for any failed scenario, please test all the possible scenarios before merging.

**Comment by sharanya-appsmith** [comment]
TBP pass multiple tags - https://github.com/appsmithorg/appsmith/actions/runs/7419317281/job/20188631044
TBP pass (@tag.All) - https://github.com/appsmithorg/appsmith/actions/runs/7419707023
TBP fail (@tags.ALL) - https://github.com/appsmithorg/appsmith/actions/runs/7419698363/job/20189685508
TBP fail (@tag.all) - https://github.com/appsmithorg/appsmith/actions/runs/7419715559
TBP fail (@tag.ALL) - https://github.com/appsmithorg/appsmith/actions/runs/7419720703

**Comment by sharanya-appsmith** [comment]
> TBP pass multiple tags - https://github.com/appsmithorg/appsmith/actions/runs/7419317281/job/20188631044
> TBP pass (@tag.All) - https://github.com/appsmithorg/appsmith/actions/runs/7419707023
> TBP fail (@tags.ALL) - https://github.com/appsmithorg/appsmith/actions/runs/7419698363/job/20189685508
> TBP fail (@tag.all) - https://github.com/appsmithorg/appsmith/actions/runs/7419715559
> TBP fail (@tag.ALL) - https://github.com/appsmithorg/appsmith/actions/runs/7419720703

**Comment by sharanya-appsmith** [comment]
Added tbp workflows with different scenarions 
cc @sarojsarab

File: .github/workflows/integration-tests-command.yml (hunk 2 above)
Comment (sharanya-appsmith) [reply .github/workflows/integration-tests-command.yml]: Changed this flow

**Comment by sarojsarab** [comment]
So the current changes are for TBP workflow, make changes in integration-tests-command.yml for ok-to-test

**Comment by sharanya-appsmith** [comment]
> So the current changes are for TBP workflow, make changes in integration-tests-command.yml for ok-to-test

TBP I haven't changed, I tested via tbp. Changes are moved to integration-tests-command.yml

File: .github/workflows/integration-tests-command.yml [hunk 3]
```diff
@@ -36,14 +36,29 @@ jobs:
         id: checkAll
         run: |
           tags="${{ github.event.client_payload.slash_command.args.named.tags }}"
-          if [[ $tags == *"@tag.All"* ]]; then
+          if [[ ($tags == *"ALL"* || $tags == *"All"*  || $tags == *"all"*) && $tags != *"@tag.All"* ]]; then
+            echo "tags_All=$tags" >> $GITHUB_OUTPUT
```
Comment (mohanarpit) [inline .github/workflows/integration-tests-command.yml:40]: Minor: Can we rename this output to `invalid_tags_all`? This is to make the check in the next step for non-empty `invalid_tags_all` to be explicit and understandable.

**Review by mohanarpit — APPROVED** [review]

**Pull request merged**